Gems/PythonAssetBuilder/Editor/Scripts/scene_api/actor_group.py

```diff
@@ -0,0 +1,499 @@
+    typename: str
+        The type name of the abstract classes to be serialized
+
+    id: UUID
+        a unique ID for the rule
+
+    Methods
+    -------
+    to_dict()
+        Converts contents to a Python dictionary
+        Adds the '$type' member
+        Adds a random 'id' member
+    """
+    def __init__(self, typename):
+        self.typename = typename
+        self.id = uuid.uuid4()
+        
+    def __eq__(self, other):
+        return self.id.__eq__(other.id)
+
+    def __ne__(self, other):
+        return self.__eq__(other) is False
+
+    def __hash__(self):
+        return self.id.__hash__()
+    
+    def to_dict(self):
+        data = self.__dict__
+        data['id'] = f"{{{str(self.id)}}}"
+        # rename 'typename' to '$type'
+        data['$type'] = self.typename
+        data.pop('typename')
+        return data
+
+class SceneNodeSelectionList(BaseRule):
+    """
+    Contains a list of node names to include (selectedNodes) and to exclude (unselectedNodes)
+
+    Attributes
+    ----------
+    selectedNodes: `list` of str
+        The node names to include for this group rule
+        
+    unselectedNodes: `list` of str
+        The node names to exclude for this group rule
+
+    Methods
+    -------
+    convert_selection(self, container, key):
+        this adds its contents to an existing dictionary container at a key position
+
+    select_targets(self, selectedList, allNodesList:list)
+        helper function to include a small list of node names from list of all the node names
+
+    to_dict()
+        Converts contents to a Python dictionary
+    """
+    def __init__(self):
+        super().__init__('SceneNodeSelectionList')
+        self.selectedNodes = []
+        self.unselectedNodes = []
+
+    def convert_selection(self, container, key):
+        container[key] = self.to_dict()
+        
+    def select_targets(self, selectedList, allNodesList:list):
+        self.selectedNodes = selectedList
+        self.unselectedNodes = allNodesList.copy()
+        for node in selectedList:
+            if node in self.unselectedNodes:
+                self.unselectedNodes.remove(node)
+
+class LodNodeSelectionList(SceneNodeSelectionList):
+    """
+    Level of Detail node selection list
+    
+    The selected nodes should be joints with BoneData
+    derived from SceneNodeSelectionList
+    see also LodRule
+
+    Attributes
+    ----------
+    lodLevel: int
+        the level of detail to target where 0 is nearest level of detail
+        up to 5 being the farthest level of detail range for 6 levels maximum
+
+    Methods
+    -------
+
+    to_dict()
+        Converts contents to a Python dictionary
+    """
+    def __init__(self):
+        super().__init__()
+        self.typename = 'LodNodeSelectionList'
+        self.lodLevel = 0
+
+    def to_dict(self):
+        data = super().to_dict()
+        data['lodLevel'] = self.lodLevel
+        return data
+
+class PhysicsAnimationConfiguration():
```

Comment:
Would it make sense to separate some of these into different files? It might make it easier for the people who maintain Physics, and Character stuff to find it and make changes.

Reply:
The actor group rules are kept together with the physics models are in physics_data.py which is a bit cleaner than keeping it all in the same module (actor_group). I am currently working on motion_group.py which is starting to show some natural common code, and Mike has already started a utils module that I will be adopting during the motion_group work